### PR TITLE
Use packaging for version comparisons.

### PR DESCRIPTION
**Pull request by QuLogic**
Because `distutils` is deprecated in Python 3.10.

- [ ] Closes #xxxx
- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask` / `isort dask`

### CODE REVIEW

File: dask/dataframe/io/parquet/arrow.py
Comment (jsignell) [inline dask/dataframe/io/parquet/arrow.py:32]: These are not equivalent. Is there a reason to prefer to use `.major`?

**Review by jsignell — APPROVED** [review]
This looks good to me - thanks so much for opening it!

**Comment by jsignell** [comment]
I found one more place where we are using `LooseVersion` in https://github.com/dask/dask/blob/main/dask/compatibility.py

**Comment by jsignell** [comment]
@jrbourbeau I assume that the added dependency is tolerable, but I'd like a second opinion.

File: dask/array/numpy_compat.py
Comment (jrbourbeau) [inline dask/array/numpy_compat.py:4]: Nitpick: It might be easier for readers to reason about if we instead `import packaging.version` and then call `packaging.version.parse(...)` directly when parsing a version string. Otherwise, folks will often need to look up where `parse_version` is coming from. Additionally, it's often difficult to use import aliases uniformly across a project.

**Review by jrbourbeau — COMMENTED** [review]
Thanks for the PR @QuLogic! Is `packaging.version` the recommended way to parse packages today? I know there's a `pkg_resources.parse_version` utility in `setuptools`, but it looks like [internally that just uses `packaging.version.parse`](https://github.com/pypa/setuptools/blob/a4dbe3457d89cf67ee3aa571fdb149e6eb544e88/pkg_resources/__init__.py#L112-L116). 

cc @jakirkham who might find this interesting

File: dask/dataframe/io/parquet/arrow.py
Comment (QuLogic) [reply dask/dataframe/io/parquet/arrow.py:32]: It's shorter, and one less parse operation to carry out.

**Comment by QuLogic** [comment]
> I found one more place where we are using `LooseVersion` in https://github.com/dask/dask/blob/main/dask/compatibility.py

Yea, I wasn't sure about that one, as nothing in `dask` appears to use it? So as public API, it might have wanted to keep the same type? Or maybe nothing external uses it and it should be removed entirely?

File: dask/array/numpy_compat.py
Comment (QuLogic) [reply dask/array/numpy_compat.py:4]: This makes many lines quite long, and `black` formats them quite poorly, so the shorter name seemed better. Also, it matches the name used in `setuptools`, in case you wanted to fall back to that when `packaging` is unavailable (it used to have its own implementation.)

**Comment by jrbourbeau** [comment]
> Yea, I wasn't sure about that one, as nothing in dask appears to use it? So as public API, it might have wanted to keep the same type? Or maybe nothing external uses it and it should be removed entirely?

We'll want to keep the variable around as we sometimes need it when we need to construct compatibility code across different versions of Python. Though you bring up a good point about this being in our public API. See https://github.com/dask/dask/pull/7824 which proposes we make `PY_VERSION` private.

**Comment by QuLogic** [comment]
> We'll want to keep the variable around as we sometimes need it when we need to construct compatibility code across different versions of Python.

Why not use `sys.version_info` for that directly though?

**Comment by jrbourbeau** [comment]
Mainly so we can conveniently/compactly handle version comparisons. For example, `PY_VERSION > "3.7"`. It also nice to have a single variable to grep for when removing version-dependent compatibility code. Though I suspect grepping for `sys.version_info` would be as effective as `PY_VERSION`

**Comment by QuLogic** [comment]
`sys.version_info > (3, 7)` seems just as compact to me. Also, if you convert `PY_VERSION` to `packaging`, then it'd have to be written `PY_VERSION > parse_version("3.7")` (or the full `packaging.version.parse` if you don't alias it).

**Review by jrbourbeau — APPROVED** [review]
Thanks @QuLogic. I pushed a few small commits which primarily just explicitly add `packaging` to CI builds. I also set version `20.0` of `packaging` as the minimum support version (which is just over 18 months old). Will merge once CI finishes

**Pull request merged**
